pages/form/TravelMethod.jsx

```diff
@@ -1,27 +1,63 @@
+import { useRouter } from "next/router";
+import ContinueButton, {
+  BackButton,
+} from "../../components/LinkButton/LinkButton";
 import { useState } from "react";
-import { Box, FormControl, Heading, Text, Flex ,  Collapse,} from "@chakra-ui/react";
+import { Box, Heading, Flex, Container } from "@chakra-ui/react";
 import Layout from "../../components/Layout/Layout";
 import useForm from "../../components/FormProvider";
-import {
-  ContinueButton,
-  BackButton,
-} from "../../components/LinkButton/LinkButton";
-import { TravelMethodButtons } from "../../components/TravelMethodButtons/TravelMethodButtons";
+import ModesOfTransportButtons from "../../components/TravelMethodButtons/TravelMethodButtons";
 import Q4Progress from "../../public/images/progress-bar/travelMethodSelection-progress-dots.svg";
+
+import { modesOfTransport } from "../../utils/constants";
 import Q4Cloud from "../../public/images/clouds/cloud-travelMethodSelection.svg";
-import { useRouter } from "next/router";
 import { sendLogs } from "../../utils/sendLogs";
-export default function Question4() {
+
+// testing data object
+
+let answ = {
+  traveledWay:[
+      { Bus: [] },
+      { Car: [] },]
+}
+
+const tranKey = (Object.keys(Object.assign({}, ...answ.traveledWay)));
+console.log(tranKey)
+// Testing works for collection initial answered saved transport mode collected in tranKey
+
+export default function TravelMethod() {
   const { answers, setAnswers } = useForm();
 
   const [transportMode, setTransportMode] = useState(
-    answers.mainTransportMode || ""
+    answers.mainTransportMode || []
+  );
+  const [count, setCount] = useState(0);
+  const [status, setStatus] = useState(
+    new Array(modesOfTransport.length).fill(false)
   );
-
-  const handleTransportMode = (e) => setTransportMode(e.target.value);
 
   const saveAnswers = () =>
     setAnswers((prev) => ({ ...prev, mainTransportMode: transportMode }));
+  
+  // handle when method button clicked
+
+  const methodClickHandler = (eventText) => {
+  
+    const ind = modesOfTransport.indexOf(eventText);
+   
+    const copy = [...status];
+    copy[ind] = !copy[ind];
+    setStatus(copy);
+
+    let selected = transportMode;
+   
+    // if (selected.includes(eventText)) {
+    //   const selected = selected.filter((mode) => mode !== eventText);
+    // } else {
+    selected = [...selected, eventText];
+    // }
```

Comment:
please remove commented-out code
```suggestion
    selected = [...selected, eventText];
```